### PR TITLE
feat: disable pre-commit manager

### PR DESCRIPTION
**Pull request by rarkins**
## Changes:

Disables pre-commit manager by default.

BREAKING CHANGE: pre-commit manager is no longer enabled by default, and must be opted into manually.

## Context:

Hostility from pre-commit maintainer.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — DISMISSED** [review]

File: lib/manager/pre-commit/readme.md [hunk 1]
```diff
@@ -1,4 +1,10 @@
-Renovate supports updating of Git dependencies within pre-commit configuration `.pre-commit-config.yaml` files or other YAML files that use the same format (via `fileMatch` configuration).
+_Important note_: The `pre-commit` manager is disabled by default and must be opted into through config.
+Renovate's approach to version updating is not fully aligned with `pre-commit upgrade` and this has caused frustration for `pre-commit`'s creator/maintainer.
+Attempts to work with the `pre-commit` project to fix these gaps have been rejected, so therefore we have chosen to disable the manager by default indefinitely.
+Please do not contact the `pre-commit` project/maintainer about any Renovate-related topic as the result is unlikely to be productive.
```
Comment (HonkingGoose) [inline lib/manager/pre-commit/readme.md:4]:
```suggestion
Please do not contact the `pre-commit` project/maintainer about any Renovate-related topic.
```

I don't think we're hitting the right tone of voice here. It feels a bit snarky to say: "Don't bother contacting them as the result is unlikely to be productive."

I think we're better off just sticking to our request: "Do not contact maintainer when you have problems with Renovate-related topics".

**Review by HonkingGoose — DISMISSED** [review]

**Comment by HonkingGoose** [comment]
For anybody wanting to know _why_ we're disabling the `pre-commit` manager by default, see this issue: https://github.com/renovatebot/renovate/issues/11166

**Review by HonkingGoose — APPROVED** [review]

File: lib/manager/pre-commit/index.ts [hunk 2]
```diff
@@ -2,5 +2,9 @@ export { extractPackageFile } from './extract';
 
 export const defaultConfig = {
   commitMessageTopic: 'pre-commit hook {{depName}}',
+  enabled: false,
   fileMatch: ['(^|/)\\.pre-commit-config\\.yaml$'],
+  prBodyNotes: [
+    'Note: The `pre-commit` manager in Renovate is not supported by the `pre-commit` maintainers or community. Please do not report any problems there, instead [create a Discussion in the Renovate repository](https://github.com/renovatebot/renovate/discussions/new) if you have any questions.',
+  ],
```
Comment (HonkingGoose) [inline lib/manager/pre-commit/index.ts:9]: I was wondering, do we need a new/adjusted unit test for this piece of code?

Comment (rarkins) [reply lib/manager/pre-commit/index.ts:9]: No, it's just config

**Review by viceice — APPROVED** [review]

**Pull request merged**